standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -7006,16 +7006,40 @@ private boolean updatePartitonColStatsInternal(Table tbl, ColumnStatistics colSt
         + " part=" + csd.getPartName());
 
     boolean ret = false;
+
+    Map<String, String> parameters;
+    List<String> partVals;
+    boolean committed = false;
+    getMS().openTransaction();
+
     try {
       if (tbl == null) {
         tbl = getTable(catName, dbName, tableName);
       }
-      ret = updatePartitionColStatsInBatch(tbl, Collections.singletonMap(csd.getPartName(), colStats),
-              validWriteIds, writeId);
+      partVals = getPartValsFromName(tbl, csd.getPartName());
+      parameters = getMS().updatePartitionColumnStatistics(colStats, partVals, validWriteIds, writeId);
+      if (parameters != null) {
+        if (transactionalListeners != null && !transactionalListeners.isEmpty()) {
+          MetaStoreListenerNotifier.notifyEvent(transactionalListeners,
+                  EventType.UPDATE_PARTITION_COLUMN_STAT,
+                  new UpdatePartitionColumnStatEvent(colStats, partVals, parameters, tbl,
+                          writeId, this));
+        }
+        if (!listeners.isEmpty()) {
```

Comment:
We usually put the non-transactional listeners after the `commitTransaction` so if there is an error there then it will not roll back the changes done in the transaction

Reply:
This is the old code so i have kept it as it is. Other methods are also using it same way.